src/elements/common/async-load/AsyncError.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+import * as React from 'react';
+
+type Props = {
+    children: React.Node,
+    component: React.ComponentType<any>,
+};
+
+type State = {
+    error: ?Error,
+};
+
+class AsyncError extends React.Component<Props, State> {
```

Comment:
Could this somehow be combined with ErrorBoundary?

Reply:
Probably, but it'll take more time. ErrorBoundary has extra logic for logging and prop forwarding, which doesn't translate directly to this simpler use case. We could tackle it in a follow-up?